assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js

```diff
@@ -48,34 +49,8 @@ function PaginatedMediaGallery({
   onInsert,
   setNextPage,
 }) {
-  // TODO(#1698): Ensure scrollbars auto-disappear in MacOS.
-  // State and callback ref necessary to recalculate the padding of the list
-  // given the scrollbar width.
-  const [scrollbarWidth, setScrollbarWidth] = useState(0);
   const refContainer = useRef();
-  const refCallbackContainer = (element) => {
-    refContainer.current = element;
-    if (!element) {
-      return;
-    }
-    setScrollbarWidth(element.offsetWidth - element.clientWidth);
-  };
-
-  // Recalculates padding so that it stays centered.
-  // As of May 2020 this cannot be achieved without js (as the scrollbar-gutter
-  // prop is not yet ready).
-  useLayoutEffect(() => {
-    if (!scrollbarWidth) {
-      return;
-    }
-    const currentPaddingLeft = parseFloat(
-      window
-        .getComputedStyle(refContainer.current, null)
-        .getPropertyValue('padding-left')
-    );
-    refContainer.current.style['padding-right'] =
-      currentPaddingLeft - scrollbarWidth + 'px';
-  }, [scrollbarWidth, refContainer]);
+  const refCallbackContainer = useAddScrollbarWidth(refContainer);
 
   const refContainerFooter = useRef();
   useIntersectionEffect(
```

Comment:
This exists in both components here (maybe not completely identical, but close to it). Can it be moved to the custom hook as well? Or maybe another shared hook?

Reply:
I think #3227 is removing most of this anyway.

Reply:
Yes, #3227 is removing this anyway, that's unrelated to this issue, too.